shell/platform/darwin/macos/framework/Source/FLEExternalTextureGL.h

```diff
@@ -0,0 +1,32 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Foundation/Foundation.h>
+
+#import "flutter/shell/platform/darwin/macos/framework/Headers/FLETexture.h"
+#import "flutter/shell/platform/embedder/embedder.h"
+
+/**
+ * Used to bridge FLETexture object and handle the texture copy request.
+ */
+@interface FLEExternalTextureGL : NSObject
+
+/**
+ * Initializes a texture adapter with |texture| return a FLEExternalTextureGL.
+ */
+- (nonnull instancetype)initWithFLETexture:(nonnull id<FLETexture>)texture;
+
+/**
+ * Accept texture buffer copy request from the flutter engine.
+ */
+- (BOOL)populateTextureWithWidth:(size_t)width
+                          height:(size_t)height
+                         texture:(nonnull FlutterOpenGLTexture*)texture;
```

Comment:
The uses of the word "texture" in this class (header and implementation) need disambiguation. The class itself is a texture, and accepts two different objects that are also textures. In this method, kind of texture is the parameter? (The type provides the answer here, but not at a call site).

Reply:
Looking more closely at the implementation, it looks like the texture being populated is the texture that is the last parameter; the current naming makes it sounds like some other internal texture is being populated from these arguments. As a general rule, you should never be naming the same parameter twice in the signature.

So unless I'm misunderstanding, this should be `populateTexture:withWidth:height:`.